src/pkg/extensions/extensions.go

```diff
@@ -0,0 +1,126 @@
+package extensions
+
+import (
+	"context"
+	"io"
+
+	"github.com/alcionai/clues"
+
+	"github.com/alcionai/corso/src/pkg/backup/details"
+	"github.com/alcionai/corso/src/pkg/logger"
+)
+
+// Extension client interface
+type CorsoItemExtension interface {
```

Comment:
> Lets keep Corso* out of our variable naming

Sounds good. 

> Nested interfaces are good for extending the set of behaviors (ie, when you need it to do foo/bar/baz, plus some extra behavior)

Thanks for the detailed overall review, first of all! 
I nested the `io.ReadCloser` so that we can have the flexibility to add extra behavior later on. For e.g. we could add `GetIdentifier()` to `CorsoItemExtension` to fetch the extension UUID/other properties. 
This could then be then used towards - 
a. Logging
b. Tracking dedicated scratch space for extensions ( uuid -> `ExtensionInfo`) sort of thing.

That being said, I don't see any current need to expose `GetIdentifier` or similar extra behavior. So I can remove `CorsoItemExtension` and just use `io.ReadCloser`.